usaspending_api/awards/management/commands/loadcontracts.py

```diff
@@ -228,34 +227,92 @@ def fetch_country_code(vendor_country_code):
         for word in code_str.split():
             query_set = query_set.filter(country_name__icontains=word)
         country_code = query_set.first()
-    
+
     return country_code
 
 
-def get_or_create_location(row):
-    country_code = fetch_country_code(row['vendorcountrycode'])
+'''
+PLACE OF PERFORMANCE
+
+CITY: placeofperformancecity
+CD: placeofperformancecongressionaldistrict
+CD: pop_cd
+COUNTRY: placeofperformancecountrycode
+ZIPCODE: placeofperformancezipcode
+STATE: pop_state_code
+STATE: statecode
+
+VENDOR
+
+CITY: city
+CD: congressionaldistrict
+CD: vendor_cd
+COUNTRY: vendorcountrycode
+ZIPCODE: zipcode
+STATE: vendor_state_code
+STATE: state
+VENDORSTREET1: streetaddress
+VENDORSTREET2: streetaddress2
+VENDORSTREET3: streetaddress3
+
+countryoforigin ????? blank in most cases anyway
+locationcode ????? blank in most cases
+'''
+
+
+def get_or_create_location(row, mode="vendor"):
+    place_of_performance_mapping = {
+        "city": row["placeofperformancecity"],
+        "congressionaldistrict": row["placeofperformancecongressionaldistrict"][2:],  # Need to strip the state off the front
+        "country": row["placeofperformancecountrycode"],
+        "zipcode": row["placeofperformancezipcode"].replace("-", ""),  # Either ZIP5, or ZIP5+4, sometimes with hypens
+        "state_code": row["pop_state_code"].split(":")[0]  # Format is VA: VIRGINIA, so we need to grab the first bit
```

Comment:
Thanks for leaving this formatting comments inline--our future colleagues will surely thank you for this